instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/AttributeInfo.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jmx.engine;
+
+/**
+ * A class holding relevant information about an MBean attribute which will be used for collecting
+ * metric values. The info comes directly from the relevant MBeans.
+ */
+class AttributeInfo {
+
+  private boolean usesDoubles;
+  private String description;
```

Comment:
do we need to worry about volatile on these? or it's always used inside a single thread executor?

Reply:
It is always used by a single thread, during MBean discovery cycle. And the cycles cannot overlap, because the next cycle is scheduled only upon completion of the previous one.